sdk/appcenter/src/main/java/com/microsoft/appcenter/http/HttpClientNetworkStateHandler.java

```diff
@@ -42,6 +42,9 @@ public synchronized ServiceCall callAsync(String url, String method, Map<String,
         Call call = new Call(mDecoratedApi, url, method, headers, callTemplate, serviceCallback);
         if (mNetworkStateHelper.isNetworkConnected()) {
             call.run();
+
+            /* Send pending call(s), if any. It can be the case if a network state update event is missing. */
+            onNetworkStateUpdated(true);
```

Comment:
This is not the right solution I think.

Reply:
I agree that we can play with network capabilities, but this might be saved as a fallback.

Reply:
But isNetworkConnected returns false if there is no event with the modern APIs, also the old APIs might work fine on older devices, the customer is on API level 23 apparently so using the new APIs.

Reply:
(also getActiveNetworkInfo is out of sync with events on some Android versions and has been deprecated, specifically on Android 8 where it can say "blocked" while we received an event for active using old APIs)

Reply:
it's not deprecated. I tested on Android 9 and yes, it's out of sync with events, but it's not an issue here.

Reply:
I'm still uncomfortable with this line. It's code executed from a specific consumer of the network state helper lib and was not needed to fix the issue. Can we revert that line?